energyID Docs (#37472)
home-assistant/home-assistant.github.io: source/_integrations/energyid.markdown

---
title: EnergyID
description: Instructions for integrating EnergyID energy management platform with Home Assistant.
ha_category:
  - Energy
  - Hub
ha_iot_class: Cloud Push
ha_domain: energyid
ha_integration_type: integration
ha_release: 2023.6
ha_config_flow: true
ha_codeowners:
  - '@JrtPec'
  - '@molier'
related:
  - docs: /integrations/energy
    title: Energy management documentation
  - url: https://www.energyid.eu/
    title: EnergyID Platform
---

> The **EnergyID** integration enables seamless transfer of energy metrics from Home Assistant to [EnergyID](https://www.energyid.eu/), a professional energy management platform for long-term tracking and analysis.

Reviewer: ```suggestion
The **EnergyID** {% term integration %} enables seamless transfer of energy metrics from Home Assistant to [EnergyID](https://www.energyid.eu/), a professional energy management platform for long-term tracking and analysis.
```

## About EnergyID

EnergyID provides tools for:

- Historical energy consumption analysis
- Solar production tracking across multiple sites
- Utility bill management and validation
- Carbon footprint reporting
- Multi-property energy comparisons

## Prerequisites

You will need:

1. Active EnergyID account with webhook permissions
2. Webhook URL from EnergyID's integration settings
3. Sensor entity in Home Assistant with valid numerical readings
4. Configured metric type in EnergyID that matches your sensor data

## How you can use this integration

- **Long-term trend analysis**: Track energy patterns over years
- **Multi-site management**: Aggregate data from multiple properties
- **etc**
{% include integrations/config_flow.md %}

> ## Configuration Options

Reviewer: ```suggestion
## Configuration options
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization) also in headings

{% configuration_basic %}
EnergyID Webhook URL:
  description: "Unique URL provided by EnergyID's webhook configuration (required)"
Home Assistant Entity ID:
  description: "Sensor entity to monitor for state changes (required)"
EnergyID Metric:
  description: "Measurement category from EnergyID's supported metrics (required)"
EnergyID Metric Kind:
  description: "Data interpretation method - cumulative/total/delta/gauge (required)"
Unit of Measurement:
  description: "Units matching EnergyID's system (kWh, m³, °C, etc.) (required)"
EnergyID Data Interval:
  description: "Data resolution (P1D, PT1H, PT15M etc.) - defaults to daily (optional)"
Upload Interval (seconds):
  description: "Minimum time between uploads - defaults to 300s (5 mins) (optional)"
{% endconfiguration_basic %}

> ## Key Features

Reviewer: ```suggestion
## Key features
```

> ### Data Handling

Reviewer: ```suggestion
### Data handling
```

- **Smart Upload Throttling**: Minimum 5-minute interval between updates
- **Value Validation**: Automatic filtering of non-numeric states
- **Connection Resilience**:
  - 3 retry attempts with exponential backoff
  - Automatic reconnection after network failures
  - Configurable retry intervals (1s, 2s, 4s)

> ### Metric Types

Reviewer: ```suggestion
### Metric types
```

- **Cumulative**: For ever-increasing counters (e.g., total kWh)
- **Total**: Interval-based consumption (e.g., daily usage)
- **Delta**: Change between measurements
- **Gauge**: Instantaneous readings (e.g., temperature)

> ### Data Resolution Options

Reviewer: ```suggestion
### Data resolution options
```

> | Interval | Description | Subscription Required |

Reviewer: ```suggestion
| Interval | Description | Subscription required |
```

|----------|-------------|-----------------------|
| PT5M     | 5-minute    | Premium               |
| PT15M    | 15-minute   | Premium               |
| PT1H     | Hourly      | Free                  |
| P1D      | Daily       | Free                  |

> ## Installation Guide

Reviewer: ```suggestion
## Installation guide
```

> ### Obtaining Webhook URL

Reviewer: ```suggestion
### Obtaining a webhook URL
```

> ### Obtaining a webhook URL

Reviewer: should this section be part of the prereqs? It looks like you need this before you add the integration to Home Assistant?

1. Log in to your EnergyID account
2. Navigate to **Integrations** → **Incoming Webhooks**
3. Create new webhook with:
   - Target record
   - Descriptive name
4. Copy the generated URL (format: `https://app.energyid.eu/integrations/WebhookIn/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`)

> ### Home Assistant Setup

Reviewer: ```suggestion
### Home Assistant setup
```

> ### Home Assistant setup

Reviewer: this should probably also go up there, or below the configuration section?

1. {% my config_flow_start title="Add EnergyID Integration" %} via Home Assistant UI
2. Paste Webhook URL when prompted
3. Select entity from dropdown list
4. Choose matching EnergyID metric type
5. Configure advanced options if needed

> ## Data Flow Management

Reviewer: ```suggestion
## Data flow management
```

> ### Update Behavior

Reviewer: ```suggestion
### Update behavior
```

> - Immediate upload on state change

Reviewer: maybe this could be more explicitly stating that the EnergyID integration is pushing data from Home Assistant to the EnergyID cloud.

> - Minimum 5-minute interval between updates (configurable)

Reviewer: how do I configure this?

- Last valid value persists during connection issues

> ### Connection Monitoring

Reviewer: ```suggestion
### Connection monitoring
```

- Automatic health checks every 15 minutes
- System log entries for connection status changes
- Visual indicator in Home Assistant UI

> ## Known Limitations

Reviewer: ```suggestion
## Known limitations
```

- **One-way synchronization**: Data flows only from Home Assistant → EnergyID
- **Premium features**: Sub-1hour intervals require EnergyID subscription
- **Entity limitations**: Each sensor requires separate webhook configuration

## Troubleshooting

> ### Common Issues

Reviewer: ```suggestion
### Common issues
```

- **`Invalid webhook URL`**: Regenerate in EnergyID and update configuration.
- **`Non-numeric state`**: Check sensor's unit_of_measurement.
- **`Connection lost`**:
  1. Verify network connectivity.
  2. Check EnergyID service status.
  3. Validate webhook URL is still active.

> ### Diagnostic Steps

Reviewer: ```suggestion
### Diagnostic steps
```

For detailed troubleshooting guidance, refer to Home Assistant's [troubleshooting documentation](/docs/configuration/troubleshooting/#enabling-debug-logging).

> ## Removing the Integration

Reviewer: ```suggestion
## Removing the integration
```

> 1. Go to {% my integrations title="**Settings** > **Devices & Services**" %}
> 2. Locate EnergyID integration card
> 3. Select **Configure**
> 4. Choose **Delete Integration**

Reviewer: ```suggestion
```
These steps are redundant. They are part of the include.

![image](https://github.com/user-attachments/assets/a8e44ddb-e52c-4a8c-b75e-dc44588a7276)

---
title: EnergyID
description: Instructions on how to integrate EnergyID into Home Assistant to send your sensor data to the EnergyID platform.
ha_category:
  - Energy

> ha_iot_class: cloud_push

Reviewer 2: ```suggestion
ha_iot_class: Cloud Push
```

## Prerequisites

1. An active account on [EnergyID](https://www.energyid.eu/).

2. A **Provisioning Key** and **Provisioning Secret** generated from your EnergyID portal. These credentials allow Home Assistant to securely connect to your account.

- For detailed instructions, refer to the [official EnergyID Home Assistant documentation](https://help.energyid.eu/en/apps/home-assistant/).

{% include integrations/config_flow.md %}

During the setup, you will be prompted for the following information:

{% configuration_basic %}
Provisioning Key:
  description: The Provisioning Key obtained from your EnergyID portal.

Provisioning Secret:
  description: The Provisioning Secret associated with your key, obtained from your EnergyID portal.
{% endconfiguration_basic %}

### Initial setup steps

1. After adding the integration, you will first be asked to enter your **Provisioning Key** and **Secret**.

>     <p class='img'><img src='/images/integrations/energyid/image-2.png' alt="Screenshot of the EnergyID connection screen in Home Assistant, asking for Provisioning Key and Secret."/></p>

Reviewer 2: We should remove unused images from the repository and ensure that no provisioning key or secret is within the screenshots. You can also consider your own provision key and secret to be leaked at this point. 

A screenshot is not a requirement, it's of course welcome if you want to have it included.

Author: @silamon that is just an example , so the user can actually see what it looks like, lengthwise etc. That key/secret have been long null & void. I quite like screenshots to show people an example, but will remove if not customary? what do you reckon?

Reviewer 2: Then we keep them in. There is still one unused image though.

Author: ah yeah i just saw it , thanks! removed it, not needed indeed

{% configuration_basic %}
Home Assistant sensor:
  description: Select the sensor entity from your Home Assistant instance whose data you want to send. The list is automatically filtered to suggest suitable numeric sensors.
{% endconfiguration_basic %}

<p class='img'><img src='/images/integrations/energyid/image-1.png' alt="Screenshot of the EnergyID configuration screen in Home Assistant, showing options to add and manage sensor mappings."/></p>

When you select a sensor, its `object_id` (the part of the entity ID after the dot) will be used as the **EnergyID Metric Key**. For example, mapping `sensor.total_active_power` will send data to EnergyID with the key `total_active_power`.

## Data updates

The EnergyID integration uses a push-based mechanism with batching:

- It listens for {% term state %} changes on your mapped sensors.
- When a sensor's value changes, the new value and timestamp are queued.
- The queued data is automatically sent to EnergyID in batches. The upload interval is determined by the policy received from EnergyID (typically every 60 seconds).

This is more efficient than traditional {% term polling %}, as it only sends data when there are new updates.

> ## Use Cases

Reviewer: ```suggestion
## Use cases
```

- Do **not** let yourself be limited by anything. No PV provider nor dongle firmware and send anything your tinkering heart desires to eid for storage and smart analysis.
- **Benchmarking & Reporting:** Utilize EnergyID's features to compare your energy usage against anonymized data from similar households and generate detailed reports.

## Troubleshooting

### Data not appearing in EnergyID

> - **Verify Mappings**: In Home Assistant, go to the EnergyID integration's configuration page and ensure your sensors are correctly mapped.

Reviewer: are the string marked in bold UI strings, or titles? If they are titles, please use headings instead of bold.
If you have a bit more context you can provide for troubleshooting, you could also use the symptom/resolution style used for troubleshooters. https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown#troubleshooting

---
title: EnergyID
description: Instructions on how to integrate EnergyID into Home Assistant to send your sensor data to the EnergyID platform.
ha_category:
  - Energy
ha_iot_class: cloud_push
ha_domain: energyid
ha_integration_type: service
ha_config_flow: true
ha_codeowners:
  - '@JrtPec'
  - '@Molier'
ha_release: 2025.12
---

> Connect your Home Assistant to [EnergyID](https://www.energyid.eu/)—a cloud platform for energy monitoring and optimization. This integration uploads your Home Assistant sensor data and provides advanced analytics and performance tracking for solar, battery, energy consumption and more.

Reviewer: ```suggestion
The **EnergyID** {% term integration %} connects your Home Assistant to [EnergyID](https://www.energyid.eu/)—a cloud platform for energy monitoring and optimization. This integration uploads your Home Assistant sensor data and provides advanced analytics and performance tracking for solar, battery, energy consumption, and more.
```

## Prerequisites

1. An active account on [EnergyID](https://www.energyid.eu/).

2. A **Provisioning Key** and **Provisioning Secret** generated from your EnergyID portal. These credentials allow Home Assistant to securely connect to your account.

- For detailed instructions, refer to the [official EnergyID Home Assistant documentation](https://help.energyid.eu/en/apps/home-assistant/).

## Configuration

> Adding EnergyID to your Home Assistant instance is done via the user interface.

Reviewer: ```suggestion
```
